### ISSUE TITLE
The .env injection for the `run` command need a fix 

### ISSUE THREAD

**Issue by jefer94** [pdm-project/pdm#1285]
# Files to reproduce

```sh
# shell
export DATABASE_URL=5 
pdm run python a.py
```

```
# .env
DATABASE_URL=postgres://user:pass@localhost:5432/database
```

```toml
# pyproject.toml
[tool.pdm.scripts]
_.env_file = ".env"
```

```py
# a.py
import os

print('print', os.environ['DATABASE_URL'])
```

# Current behavior

```sh
5
```

# Expected behavior

```sh
postgres://user:pass@localhost:5432/database
```

PR https://github.com/pdm-project/pdm/pull/1284

**Comment by frostming** [comment]
Which is the correct behavior depends on how you think about it.

@pgjones has a [different opinion on this](https://github.com/pdm-project/pdm/issues/1235).

**Comment by jefer94** [comment]
The standard generally is, load the `.env` if it exists, and I you are in production, get the environment thought of cloud provider I think, maybe aproach be add a as it

```toml
[tool.pdm.scripts.env] # or [tool.pdm.env] 
env_file = ".env"
prefer = "dotenv" # or "system"
```

The configuration section should be `[tool.pdm.scripts.env]` or `[tool.pdm.env]`, the second way allow use the same environment to `pdm run` and `pdm shell`

Also, `pipenv` overwrite the existing environment variable with the value of `.env`

The new aproach question to the user what prefer, and keep the behavior of `pipenv` to make more easier the migration from it, also, the consecuense of this way ends with if you install `pdm` in a project was `pipenv` the setting per default should be set to `env_file = ".env"` and `prefer = "dotenv"`, else ask to the user what prefer

What do you think? 🤔, the last way to set the `env_file` should keep their compatibility per a time

**Comment by jefer94** [comment]
Also if a developer use `pdm` with `gitpod` the `DATABASE_URL` generally is set, if that developer want to work with a custom environment variable, coping the content of `.env.example`, the way to load their custom variable is more harder than prefer the value of `.env` if it exists

**Comment by pgjones** [comment]
dotenv does provide an override open (defaulting to False) to allow this to be customized. The same could be used here as a solution to both assumptions. 

I still believe an env file should not override an environment variable by default, see also ["By default, load_dotenv doesn't override existing environment variables."](https://saurabh-kumar.com/python-dotenv/)

**Comment by jefer94** [comment]
Which should be the approach related to set override in the config file? I think also this depends if the command is being executed in `development`, `ci` or `production` in each environment is more convenient to have one preference or other, `daphne` is prefer run without override, `python manage.py runserver ` is useful in the development environment and is used by junior developers that don't understand how the project works and is prefer setup it with override

Maybe a good option should be considering this is implemented `test = {cmd = "pytest", env_file=".env"}` implement a

```toml
# The way 1, choice the environment using the same style than `test = {cmd = "pytest", env_file=".env"}`
[tool.pdm.scripts]
test1 = {cmd = "pytest", environment="prod"}
test2 = {cmd = "pytest", environment="dev"}

[tool.pdm.environments]
prod = {override=true, env_file=".env"}
dev = {override=false, env_file=".env"}
```

```toml
# The way 2, read first the environments and infer the configuration
[tool.pdm.scripts.prod]
test1 = "pytest"

[tool.pdm.scripts.dev]
test2 = "pytest"

[tool.pdm.environments]
prod = {override="true", env_file=".env"}
dev = {override="false", env_file=".env"}
```

**Comment by pgjones** [comment]
I think it is unnecessary for multiple environment files and that by definition the variables in the environment that the code runs in should take precedence - to me this is the point of environment variables they exist in the environment the code runs in to configure it.

I would recommend,
```
[tool.pdm.scripts]
script_cmd = {cmd = "...", env_file="...", override = false}
```
as a compromise with the default staying as it is now.

**Comment by frostming** [comment]
> I would recommend,
> 
> ```
> [tool.pdm.scripts]
> script_cmd = {cmd = "...", env_file="...", override = false}
> ```
> 
> as a compromise with the default staying as it is now.

This is not clear what `override` is to override, especially when `env_file` is not given, and that requires more validation logic.

Taking advantage of the TOML syntax, I would recommend:

```toml
[tool.pdm.scripts]
script_cmd = {cmd = "...", env_file.override="..."}
```

**Comment by pgjones** [comment]
Makes sense, I'll open a PR

**Comment by pgjones** [comment]
See #1299

**Comment by frostming** [comment]
Closed by #1299
